Yelp/nrtsearch
# DeleteIndexBackup - also delete backup data from _version folder

## Changes

* Delete old index backup data and metadata from `_version` folder
* Changed the structure `DeleteIndexBackupResponse` to be more descriptive. Rather than returning list of hashes, break it down to 4 lists: `deletedResourceDataHashes`,  `deletedResourceMetadataHashes`, `deletedDataVersions` and `deletedMetadataVersions`. See comments for description of those fields.

## Manual testing

Start (and restore) an index

```
curl -XPOST localhost:10003/v1/start_index -d '{
  "indexName" : "talk_v1",
   "mode": "PRIMARY",
   "restore" : {
     "serviceName" : "nrtsearch-generic",
     "resourceName": "talk_v1"
   }
}'
```

Create a few backups

```
curl -XPOST localhost:10003/v1/backup_index -d '{
	"indexName": "talk_v1", 
	"serviceName": "nrtsearch-generic-jblaszyk",
	"resourceName": "talk_v1"
}'
```

Call the delete index backup endpoint.

```
curl -XPOST localhost:10003/v1/delete_index_backup -d '{
	"indexName": "talk_v1", 
	"serviceName": "nrtsearch-generic-jblaszyk", 
	"resourceName": "talk_v1", 
	"nDays": 5
}'
```
Response - see: old, non-deleted versions now getting deleted.
```
{
    "deletedResourceDataHashes":["774f7fe3-f894-4ef6-bb6b-efe0abe97548"],
    "deletedResourceMetadataHashes":["408d7dac-dad8-4ceb-bca9-38775bd63624"],
    "deletedDataVersions":["0","1","2","3","4","5","6","7","8"],
    "deletedMetadataVersions":["0","1","2","3","4","5","6","7","8"]
}
```
The old data from `_version` folder was deleted as expected.
